File: src/datagen/main/java/mekanism/common/patchouli/GuideCategory.java
package mekanism.common.patchouli;

import java.util.Locale;
import mekanism.api.text.IHasTranslationKey;
import mekanism.common.Mekanism;
import net.minecraft.util.Util;

/**
 * Created by Thiakil on 19/05/2020.
 */
public enum GuideCategory {
    ITEMS,
    ITEMS_METAL_AND_ORE,
    ITEMS_GEAR,
    ITEMS_UPGRADES,
    BLOCKS,
    MULTIBLOCKS,
    ORE_PROCESSING,
    CHEMICALS
    ;

    public final String id;
    private final String translationKey;

    GuideCategory() {
        this.id = name().toLowerCase(Locale.ROOT);
        this.translationKey = Util.makeTranslationKey("guidebook", Mekanism.rl("category."+id));
    }

    public String getTranslationKeyName() {
        return translationKey+".name";
    }

    public String getTranslationKeyDescription() {
        return translationKey+".description";
    }
}

File: src/datagen/main/kotlin/mekanism/patchouli/GuideCategory.kt
package mekanism.patchouli

import mekanism.patchouli.dsl.IGuideCategory
import java.util.Locale
import mekanism.common.Mekanism
import net.minecraft.util.Util

enum class GuideCategory : IGuideCategory {
    ITEMS,
    ITEMS_METAL_AND_ORE,
    ITEMS_GEAR,
    ITEMS_UPGRADES,
    BLOCKS,
    MULTIBLOCKS,
    ORE_PROCESSING,
    CHEMICALS,
    LIQUIDS
    ;

    override val id: String = name.toLowerCase(Locale.ROOT)
    private val translationKey: String = Util.makeTranslationKey("guidebook", Mekanism.rl("category.$id"))
    override val translationKeyName: String = "$translationKey.name"
    override val translationKeyDescription: String = "$translationKey.description"

}